docs/examples/soroban/invoke_contract_functions.md

```diff
@@ -169,45 +169,25 @@ keypair2 =
 
 address_type = SCAddress.new(public_key)
 address = SCVal.new(address: address_type)
-args = [address, SCVal.new(u128: %{hi: 0, lo: 2})]
-
-function_args =
-  HostFunctionArgs.new(
-    type: :invoke,
-    contract_id: contract_id,
-    function_name: function_name,
-    args: args
-  )
-
-auth_invocation =
-  AuthorizedInvocation.new(
-    contract_id: contract_id,
-    function_name: function_name,
-    args: args,
-    sub_invocations: []
-  )
-
-# Nonce increment by 1 each successfully contract call
-address_with_nonce = AddressWithNonce.new(address: address_type, nonce: 0)
-
-contract_auth =
-  ContractAuth.new(
-    address_with_nonce: address_with_nonce,
-    authorized_invocation: auth_invocation
-  )
-  |> ContractAuth.sign(secret_key)
-
-function = HostFunction.new(args: function_args, auth: [contract_auth])
-
-invoke_host_function_op =
-  InvokeHostFunction.new(functions: [function], source_account: public_key_2)
+
+args =
+  SCVec.new([
+    contract_address,
+    function_name,
+    address,
+    SCVal.new(u128: %{hi: 0, lo: 2})
+  ])
+
+host_function = HostFunction.new(invoke_contract: args)
+
+invoke_host_function_op = InvokeHostFunction.new(host_function: host_function)
 
 source_account = Stellar.TxBuild.Account.new(public_key_2)
-{:ok, seq_num} = Accounts.fetch_next_sequence_number(public_key)
+{:ok, seq_num} = Accounts.fetch_next_sequence_number(public_key_2)
```

Comment:
IMO we should replace `public_key_2` with a more descriptive variable name. Same with `public_key_1`